src/main/java/org/tugraz/sysds/runtime/instructions/spark/RandSPInstruction.java

```diff
@@ -699,7 +859,84 @@ public GenerateRandomBlock(long rlen, long clen, int brlen, int bclen, double sp
 		}
 	}
 
-	private static class GenerateSequenceBlock implements PairFunction<Double, MatrixIndexes, MatrixBlock> 
+	private static class GenerateRandomTensorBlock implements PairFunction<Tuple2<TensorIndexes, Long>, TensorIndexes, TensorBlock>
+	{
+		private static final long serialVersionUID = -512119897654170462L;
+
+		private ValueType _vt;
+		private int[] _dims;
+		private int[] _blen;
+		private double _sparsity;
+		private String _min;
+		private String _max;
+		private String _pdf;
+		private String _pdfParams;
+
+		public GenerateRandomTensorBlock(ValueType vt, int[] dims, int[] blen, double sparsity, String min, String max,
+		                           String pdf, String pdfParams) {
+			_vt = vt;
+			_dims = new int[dims.length];
+			Array.copy(dims, 0, _dims, 0, dims.length);
+			_blen = blen;
+			_sparsity = sparsity;
+			_min = min;
+			_max = max;
+			_pdf = pdf;
+			_pdfParams = pdfParams;
+		}
+
+		@Override
+		public Tuple2<TensorIndexes, TensorBlock> call(Tuple2<TensorIndexes, Long> kv)
+				throws Exception
+		{
+			//compute local block size:
+			TensorIndexes ix = kv._1();
+			// TODO: accurate block size computation
+			int[] blockDims = new int[_dims.length];
+			blockDims[0] = UtilFunctions.computeBlockSize(_dims[0], ix.getIndex(0), _blen[0]);
+			int clen = 1;
+			for (int i = 1; i < _dims.length; i++) {
+				blockDims[i] = UtilFunctions.computeBlockSize(_dims[i], ix.getIndex(i), _blen[i]);
+				clen *= blockDims[i];
+			}
+			long seed = kv._2;
+
+			TensorBlock tb = new TensorBlock(_vt, blockDims);
+			// TODO implement sparse support
+			tb.allocateDenseBlock();
+			if (!_min.equals(_max)) {
+				if (_vt == ValueType.STRING) {
+					throw new DMLRuntimeException("Random string data can not be generated for tensors.");
+				}
+				MatrixBlock blk = new MatrixBlock();
+				RandomMatrixGenerator rgen = LibMatrixDatagen
+						.createRandomMatrixGenerator(_pdf, blockDims[0], clen, blockDims[0], clen, _sparsity,
+								Double.parseDouble(_min), Double.parseDouble(_max), _pdfParams);
+				blk.randOperationsInPlace(rgen, null, seed);
```

Comment:
I fixed it, but I actually did this on purpose since I use two tabs for each newline if I went one level deeper (I will do this differently now).
Basically in this case the first two tabs indent to show that `.creaetRandomMatrixGenerator` is called here and on the next line I have 4 tabs to show that `Double.parseDouble(_min),...` is a continuation of the parameters for `.createRandomMatrixGenerator`, so to make a visual distinction between a method call on LibMatrixDatagen and a method parameter I went for 4 indents.